biggraphite/drivers/elasticsearch.py

```diff
@@ -286,20 +285,116 @@ def delete_directory(self, name):
+        if isinstance(value, bg_glob.AnySequence):
+            return None, None  # No constrain
+        elif isinstance(value, six.string_types):
+            return 'term', value
+        elif isinstance(value, bg_glob.CharNotIn):
+            return 'regexp', '[^' + ''.join(value.values) + ']'
+        elif isinstance(value, bg_glob.CharIn):
+            return 'regexp', '[' + ''.join(value.values) + ']'
+        elif isinstance(value, bg_glob.SequenceIn):
+            return 'terms', value.values
+        elif isinstance(value, bg_glob.AnyChar):
+            return 'wildcard', '?'
+        else:
+            raise GlobError("Unhandled type '%s'" % value)
+
+    @staticmethod
+    def __parse_complex_component(component):
+        def is_wildcard_sub_component(sub_component):
+            return any([isinstance(sub_component, bg_glob.AnySequence),
+                        isinstance(sub_component, bg_glob.AnyChar),
+                        isinstance(sub_component, six.string_types)
+            ])
+
+        if all(is_wildcard_sub_component(sub_c) for sub_c in component):
+            return _ElasticSearchAccessor.__parse_wildcard_component(component)
+        return _ElasticSearchAccessor.__parse_regexp_component(component)
+
+    def __glob_names(self, glob, aggregate_dirs=False):
+        # TODO really implement it
+        if aggregate_dirs == True:
+            return []
+
+        if glob == "":
+            return []
+
+        search = Search().using(self.client).index(self._index).source('name')
+        # TODO Only if no globstars !!!
+        # en attendant, we assert no globstars
+        search = search.filter('term', length=glob.count("."))
+
+        components = self.__glob_parser.parse(glob)
+
+        if self.__glob_parser.is_fully_defined(components):
+            search = search.filter('term',**{"name" : ".".join(self._components_from_name(glob))})
+        else:
+            for i, c in enumerate(components):
+                if len(c) == 1:
+                    type, value = self.__parse_simple_component(c)
+                else:
+                    type, value = self.__parse_complex_component(c)
+
+                if type:
+                    search = search.filter(type,**{"p%d.keyword"%i : value})
+
+        log.debug(search.to_dict())
+        results = [ h.name for h in search.execute() ]
         results.sort()
         return results
 
     def glob_metric_names(self, glob):
         """See the real Accessor for a description."""
         super(_ElasticSearchAccessor, self).glob_metric_names(glob)
-        return iter(self.__glob_names(self._metric_names, glob))
+        return iter(self.__glob_names(glob))
 
     def glob_directory_names(self, glob):
         """See the real Accessor for a description."""
         super(_ElasticSearchAccessor, self).glob_directory_names(glob)
-        return iter(self.__glob_names(self._directory_names, glob))
+        return iter(self.__glob_names(glob, aggregate_dirs=True))
```

Comment:
for elasticsearch, metrics and name will be very different, so using __glob_names will not make much sense.

Reply:
Done